parse/lex.go

```diff
@@ -249,23 +251,46 @@ func defaultStateFn(s *state, e emitter) stateFn {
 	return defaultStateFn
 }
 
-// TODO: implement me w/ test cases :-)
+// TODO: handle error case. panic isn't enough
+//
 // NumberStateFn scans an alphanumeric. ex) 123, 4001, 232
 // After reading Number, it returns DefaultStateFn.
+// number = { decimal_digit }
 func numberStateFn(s *state, e emitter) stateFn {
+	if r := s.peek(); !(r == '+' || r == '-' || ('0' <= r && r <= '9')) {
+		panic("Invalid Function call : numberStateFn")
```

Comment:
How about emit Illegal token (plz refer `token/token.go`)?? I think this is where Illigal token for.

If we panic program shut down and we can't find the reason of why the program shut down. plus if we emit illegal token instead of panic we can pass the responsibility to gracefully shutting down

Reply:
thx for review!
Before fixing my code, I need some advice as beginner!

1) I leave that issue in TODO list because we don't have err message section in state struct.
even we shut down program gracefully via 'Illegal token',  how can i leave error message?

2) do we really need to check condition check? 
Apparently, doing double-check would become more safe. 
but I found out even Go SDK lex.go don't check twice time.

forgive my innocent lol

Reply:
1. I think you can leave a message using `Token.Val`, because token type is illegal we can manage illegal reason with value property
2. don't understand 'check condition check', which code?

Reply:
confirmed! appreciate u  :D